clients/unshaded/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -453,9 +453,16 @@ private TachyonByteBuffer readLocalByteBuffer(int blockIndex, long offset, long
    */
   TachyonByteBuffer readRemoteByteBuffer(ClientBlockInfo blockInfo) {
     // We call into the remote block in stream class to read a remote byte buffer
-    ByteBuffer buf =
-        RemoteBlockInStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0, blockInfo.length,
-            mTachyonConf);
+    // TODO: Avoid using this constructor, investigate the right way to read remote buffer here.
+    RemoteBlockInStream inStream = new RemoteBlockInStream(new TachyonFile(null, -1, null),
+        ReadType.NO_CACHE, -1, null, null, true);
+    ByteBuffer buf
+        = inStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0, blockInfo.length, mTachyonConf);
```

Comment:
I see. We should have a copy here. 
A side note: I don't know if we can use the Netty's buffer pool to allocate the client buffer. It could be more efficient.